plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,919 @@
+        sys_id = ''
+
+        output_lines = [
+        l for l in cntrl_info.split("\n")]
+
+        for line_num in range(len(output_lines)):
+            cur_line = output_lines[line_num]
+            tmp = cur_line.split(":")
+            if tmp[0].strip() == 'Controllers found':
+                total_cntrl = int(tmp[1].strip())
+
+
+            for cntrl in range(total_cntrl):
+                cntrl_num = 'Controller %d' %(cntrl + 1)
+                if tmp[0].strip() == cntrl_num :
+                    cntrl_info = tmp[2].split(",")
+                    controllerInfo['Status'] = cntrl_info[0].strip()
+                    controllerInfo['Slot'] = cntrl_info[1].strip()
+                    controllerInfo['Mode'] = cntrl_info[2].strip()
+                    controllerInfo['Name'] = cntrl_info[3].strip()
+                    controllerInfo['Serial Number'] = cntrl_info[4].strip()
+                else:
+                    continue
+                if controllerInfo['Mode'] == "HBA":
+                    continue
+                else:
+                    getconfig_lds_info = self._arcconf_exec([ "getconfigJSON", str(cntrl+1)], COMPONENT[5], False, False)
+                    sys_id = str(cntrl+1) 
+                    cntrl_num = str(cntrl+1)
+                    cnt = int(cntrl + 1)
+
+                    getconfig_lds_info_json = getconfig_lds_info.split("\n")[1]
+                    decoded_json = json.loads(getconfig_lds_info_json)
+                    
+
+                    try:
+                        if 'LogicalDrive' in decoded_json['Controller']: 
+                            for ld in range(len(decoded_json['Controller']['LogicalDrive'])):
+                                ld_num = decoded_json['Controller']['LogicalDrive'][ld]['logicalDriveID']
+                                ld_name = decoded_json['Controller']['LogicalDrive'][ld]['name']
+                                pool_id = '%s:%s' %(cntrl_num, ld_num)
+                                ld_info = decoded_json['Controller']['LogicalDrive'][ld]
+                                lsm_vol = Arcconf._arcconf_ld_to_lsm_vol(ld_info, pool_id, sys_id, cnt, str(ld_num), ld_name)
+                                lsm_vols.append(lsm_vol)
+                    except DeviceNotFoundError:
+                        pass
+
+        return search_property(lsm_vols, search_key, search_value)
+
+
+
+
+    @staticmethod
+    def _arcconf_disk_to_lsm_disk(arcconf_disk, sys_id, ctrl_num, flag_free=False):
+        disk_id = arcconf_disk['serialNumber'].strip()
+        
+        disk_name = "%s" % (arcconf_disk['model'])
+        disk_type = _disk_type_of(arcconf_disk)
+        blk_size = int(arcconf_disk['fsaBytesPerBlock'])
+        blk_count = int(arcconf_disk['fsaNumBlocks'])
+
+        status = _disk_status_of(arcconf_disk, flag_free)
+        disk_reported_slot = arcconf_disk['fsaSlotNum']
+        disk_channel = arcconf_disk['channelID']
+        disk_device = arcconf_disk['deviceID']
```

Comment:
Try to add a global decorator like we did in hpsa plugin `_handle_errors`, so that key error here could be treated as
`PLUGIN_BUG`.